webhook.go

```diff
@@ -103,14 +104,42 @@ func main() {
 		os.Exit(1)
 	}
 
-	if *debug {
+	if *debug || *logPath != "" {
 		*verbose = true
 	}
 
 	if len(hooksFiles) == 0 {
 		hooksFiles = append(hooksFiles, "hooks.json")
 	}
 
+	addr := fmt.Sprintf("%s:%d", *ip, *port)
+
+	// Open listener early so we can drop privileges.
+	ln, err := net.Listen("tcp", addr)
+	if err != nil {
+		log.Printf("error listening on port: %s", err)
```

Comment:
Should we move logger settings before this line.

```
log.SetPrefix("[webhook] ")
log.SetFlags(log.Ldate | log.Ltime)
```

Reply:
Good point.  I ended up refactoring the startup logging and added a small log queue, which should solve your concern and fix some "chicken and egg" problems.  I left the initial exit logging as-is since those are issues with invalid flag usages.  Once we start interacting with the OS (port listening, etc), we use `log.*` instead of `fmt.*`.

Comments?

Reply:
Looks good :-)